File: leetcode/409.cpp
class Solution {
public:
	int longestPalindrome(string s) {
		int o = 0;
		bool isSingle = false;
		int t;
		vector<int> v(256, 0);
		for (int i = 0; i < s.size(); ++i)
		{
			++v[s[i]];

		}
		for (int i = 0; i < 256; ++i) {

			if (v[i] % 2)
				isSingle = true;
			o += 2 * (v[i] / 2);
		}
		if (isSingle) ++o;
		return o;
	}
};

File: leetcode/409.java
class Solution {
    public int longestPalindrome(String s) {
        boolean hasCenter = false;
        int ret = 0;
        int[] count = new int[256];
        for (int i = 0; i < s.length(); i++) {
            count[s.charAt(i)]++;
        }

        for (int i : count) {
            ret += (i / 2) * 2;
            if (i % 2 == 1) hasCenter = true;
        }

        if (hasCenter == true)
            ret++;

        return ret;
    }
}